Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,422 @@
+/*global define*/
+define([
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/clone',
+        '../Core/Color',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/destroyObject',
+        '../Core/Matrix4',
+        '../Renderer/Buffer',
+        '../Renderer/DrawCommand',
+        '../Renderer/WebGLConstants',
+        '../ThirdParty/when',
+        './Model',
+        './SceneMode'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        clone,
+        Color,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        destroyObject,
+        Matrix4,
+        Buffer,
+        DrawCommand,
+        WebGLConstants,
+        when,
+        Model,
+        SceneMode) {
+    "use strict";
+
+    var LoadState = {
+        NEEDS_LOAD : 0,
+        LOADING : 1,
+        LOADED : 2,
+        FAILED : 3
+    };
+
+    /**
+     * A 3D model instance collection. All instances reference the same underlying model, but have unique
+     * per-instance properties like model matrix, pick id, etc.
+     *
+     * @alias ModelInstanceCollection
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} options.url The url to the .gltf file.
+     * @param {String} [options.cacheKey] The cache key for the model.
+     * @param {Object} [options.headers] HTTP headers to send with the request.
+     * @param {Boolean} [options.show=true] Determines if the collection will be shown.
+     * @param {Boolean} [options.allowPicking=false] When <code>true</code>, each glTF mesh and primitive is pickable with {@link Scene#pick}.
+     * @param {Boolean} [options.asynchronous=true] Determines if model WebGL resource creation will be spread out over several frames or block until completion once all glTF files are loaded.
+     * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Draws the bounding sphere for the collection.
+     * @param {Boolean} [options.debugWireframe=false] For debugging only. Draws the instances in wireframe.
+     */
+    var ModelInstanceCollection = function(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.url)) {
+            throw new DeveloperError('options.url is required.');
+        }
+        //>>includeEnd('debug');
+
+        this._model = undefined;
+        this._boundingSphere = undefined;
+        this._instancingEnabled = false;
```

Comment:
Perhaps `_instancingSupported`.  I had to look at all of the code to realize what this is for.